# Part 4 - Regressions

Serena Weimer

Data Analytics

December 2, 2020

In [1]:
library(dplyr)
library(tidyverse)
library(WDI)
library(plm)
library(estimatr)
library(lmtest)
library(corrplot)
library(stargazer)
library(RColorBrewer)
library(Hmisc)
#install.packages("car")
library(sandwich)
library(car)

Warning message:
"package 'dplyr' was built under R version 3.6.3"
Attaching package: 'dplyr'

The following objects are masked from 'package:stats':

    filter, lag

The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union

Warning message:
"package 'tidyverse' was built under R version 3.6.3"-- Attaching packages --------------------------------------- tidyverse 1.3.0 --
v ggplot2 3.3.2     v purrr   0.3.2
v tibble  3.0.4     v stringr 1.4.0
v tidyr   1.1.2     v forcats 0.4.0
v readr   1.3.1     
Warning message:
"package 'ggplot2' was built under R version 3.6.3"Warning message:
"package 'tibble' was built under R version 3.6.3"Warning message:
"package 'tidyr' was built under R version 3.6.3"-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
Warning message:
"package 'WDI' was built under R version 3.6.3"Warning message:
"package 'plm' w

In [2]:
#connecting to my path
setwd('C:/Users/seren/OneDrive/Desktop/SustainDevel_project/')
#calling my csv file and naming it data, and printing it
data1 <- read.csv('alldata_final.csv')
#droping columns
data = subset(data1, select = -c(X,GDP_Growth) )
data

Country     Code Year GHG    GDP_PerCap AvgTotYrs_School Unemploy
1    Afghanistan AFG  2003  16.78  190.6838   2.4                 NA  
2    Afghanistan AFG  2004  16.35  211.3821   2.5                 NA  
3    Afghanistan AFG  2005  17.09  242.0313   2.6                 NA  
4    Afghanistan AFG  2006  20.74  263.7337   2.7                 NA  
5    Afghanistan AFG  2007  24.78  359.6932   2.9                 NA  
6    Afghanistan AFG  2008  31.39  364.6607   3.0                 NA  
7    Afghanistan AFG  2009  37.72  438.0760   3.1                 NA  
8    Afghanistan AFG  2010  44.75  543.3030   3.2                 NA  
9    Afghanistan AFG  2011  58.62  591.1628   3.3                 NA  
10   Afghanistan AFG  2012  67.05  641.8715   3.4                 NA  
11   Afghanistan AFG  2013  76.29  637.1655   3.5                 NA  
12   Afghanistan AFG  2014  86.60  613.8567   3.5                 NA  
13   Afghanistan AFG  2015  93.54  578.4664   3.6                 NA  
14   Afghanistan AFG  2016  96.59  547.2281   3.6                 NA  
15   Albania     ALB  2003  10.09 1846.1188   9.0             16.684  
16   Albania     ALB  2004  10.15 2373.5798   9.1             16.586  
17   Albania     ALB  2005   9.92 2673.7873   9.1             16.460  
18   Albania     ALB  2006   8.75 2972.7433   9.2             16.196  
19   Albania     ALB  2007   9.06 3595.0372   9.2             15.966  
20   Albania     ALB  2008   8.53 4370.5401   9.2             13.060  
21   Albania     ALB  2009   8.64 4114.1401   9.3             13.674  
22   Albania     ALB  2010   8.85 4094.3503   9.3             14.086  
23   Albania     ALB  2011   9.29 4437.1429   9.3             13.481  
24   Albania     ALB  2012   8.95 4247.6300   9.6             13.376  
25   Albania     ALB  2013   9.01 4413.0609   9.7             15.866  
26   Albania     ALB  2014   9.51 4578.6320   9.7             17.490  
27   Albania     ALB  2015   9.30 3952.8012   9.7             17.080  
28   Albania     ALB  2016   9.19 4124.0557  10.0             15.220  
29   Algeria     DZA  2003 138.30 2103.4131   6.5             23.720  
30   Algeria     DZA  2004 142.19 2609.9456   6.7             17.650  
...  ...         ...  ...  ...    ...        ...              ...     
2673 Yemen,Rep.  YEM  2015  25.76 1395.4396  3.0              13.395  
2674 Yemen,Rep.  YEM  2016  23.57 1033.7330  3.0              13.307  
2675 Zambia      ZMB  2003 474.63  435.4581  6.1              14.836  
2676 Zambia      ZMB  2004 488.47  538.5915  6.2              15.380  
2677 Zambia      ZMB  2005 565.59  702.7409  6.3              15.900  
2678 Zambia      ZMB  2006 482.32 1047.9192  6.4              12.853  
2679 Zambia      ZMB  2007 452.18 1124.2906  6.4              10.183  
2680 Zambia      ZMB  2008 470.58 1394.0006  6.5               7.930  
2681 Zambia      ZMB  2009 412.12 1159.9078  6.5              10.506  
2682 Zambia      ZMB  2010 459.88 1489.4593  6.6              13.190  
2683 Zambia      ZMB  2011 431.87 1672.9083  6.7              10.343  
2684 Zambia      ZMB  2012 454.49 1763.0727  6.7               7.850  
2685 Zambia      ZMB  2013 467.69 1878.9070  6.8               8.618  
2686 Zambia      ZMB  2014 440.08 1763.0573  6.9               9.340  
2687 Zambia      ZMB  2015 472.34 1337.7961  6.9              10.105  
2688 Zambia      ZMB  2016 493.99 1280.5789  7.0              10.882  
2689 Zimbabwe    ZWE  2003  64.68  478.0074  6.8               4.733  
2690 Zimbabwe    ZWE  2004  64.87  482.9984  6.8               4.390  
2691 Zimbabwe    ZWE  2005  65.05  476.5553  6.8               4.474  
2692 Zimbabwe    ZWE  2006  63.96  447.8549  6.8               4.496  
2693 Zimbabwe    ZWE  2007  64.45  431.7872  7.0               4.529  
2694 Zimbabwe    ZWE  2008  62.88  356.6933  7.0               4.628  
2695 Zimbabwe    ZWE  2009  63.54  771.5988  7.2               5.009  
2696 Zimbabwe    ZWE  2010  66.54  948.3319  7.3               5.217  
2697 Zimbabwe    ZWE  2011  6

In [3]:
#running basic ols regression
base <- plm(log(GDP_PerCap) ~ GHG + AvgTotYrs_School + MedHighTech + Unemploy + FemLF_Partici + Gini + NatDis_Death + Renewables + Life_Expect, data = data, index = c("Code", "Year"), model = "within")
#running a regression with oneway fixed effects on my panel data 
base1 <- lm(log(GDP_PerCap) ~ GHG + AvgTotYrs_School + MedHighTech + Unemploy + FemLF_Partici + Gini + NatDis_Death + Renewables + Life_Expect, data = data)
#running a regression with twoway fixed effects on my panel data
regress2 <- plm(log(GDP_PerCap) ~ GHG + AvgTotYrs_School + MedHighTech + Unemploy + FemLF_Partici + NatDis_Death + Gini + Renewables + Life_Expect, data = data, index = c("Code", "Year"), effect= "twoways", model = "within")

#regression with clustered standard errors by years
regress3 <- plm(log(GDP_PerCap) ~ GHG + AvgTotYrs_School + MedHighTech + Unemploy + FemLF_Partici + NatDis_Death + Gini + Renewables + Life_Expect, data = data, index = c("Code", "Year"), effect= "twoways", model = "within")
clust <- vcovHC(base1, type = "HC1", cluster = "time")
clust1 <- sqrt(diag(clust))
#https://cran.r-project.org/web/packages/stargazer/vignettes/stargazer.pdf
#clustered standard error grouped by years

In [5]:
#putting regression into professional table
stargazer(base1, base, regress2, regress3, type = "text",  se = list(NULL, NULL, NULL, clust1), single.row = FALSE, align=TRUE,
    dep.var.labels=c("GDP Per Capita"), covariate.labels=c("Greenhouse Gas Emissions","Average Total Years of School",
    "Meduim/High Tech Industry","Umemployed","Female Labor Force Participation","Gini Index",
    "Natural Disaster Deaths", "Renewable Energy","Life Expectancy Rate"), no.space=TRUE, 
    title = "Table 2: Results", notes = "()std.error", out = "C:/Users/seren/OneDrive/Desktop/SustainDevel_project/table2.html")

#https://www.jakeruss.com/cheatsheets/stargazer/#make-an-addition-to-the-existing-note-section


Table 2: Results
                                                                        Dependent variable:                                       
                                 -------------------------------------------------------------------------------------------------
                                                                          GDP Per Capita                                          
                                           OLS                                             panel                                  
                                                                                           linear                                 
                                           (1)                      (2)                      (3)                     (4)          
----------------------------------------------------------------------------------------------------------------------------------
Greenhouse Gas Emissions                -0.00004*                

# Exploratory Analysis

The most basic regression preformed in this study uses the Ordinary Least Squares (OLS) method with panel data that had 887 observations. Looking at the r-squared, it can be seen that these variables account for approximately 78% of growth. Further, six variables were found to be significant. We can be 99% confident that Average total years of school, the Medium High Tech Industry Value Added, the Female Labor Force Participation Rate, and Life expectancy rate are in the 1% significance level. Where as, Greenhouse Gas Emissions fall within the 10% significance level with a 90% confidence interval.

The second regression ran is the OLS method with a one way country fixed effects. It was quite interesting to find that all of my variables were significant in the 95% and 99% confidence intervals which account for approximately 59% of GDP. The third regression run uses the OLS two way fixed effects model. The two-way fixed effects control for differneces in predictors for both time and country. My hypothesis was that this would have been the best working model, however these results held variables that did not have all significant p-values. With this model, we can be 99% confident that with Greenhouse Gas Emissions, Unemployment, the Gini Index, and renewable energy we can reject the null hypothesis. Furhter, the final regression ran invloved the OLS method with two-way fixed effects and a clustered standard error. I ran this regression to see if how my results changed from regression 3 when considering the correlation between observations. However, my results did not change significantly. The biggest change that occued was the Life expectancy variable originally had a 5% significance became insignificant.  

After considering all four regression my preffered model is regression 2, due to the number of significant findings. The coefficents can be interpreted to see the effect these variables individually have on the growth of GDP Per capita, ceteris parabus. Specifically, an increase of one unit of Greenhouse Gas Emissions will increase GDP by .02%. For every year addded to the average total years of schooling, GDP will increase by 21%. Further for every perecent increase in the medium and high tech industry, gdp growth will incrase by .6% For every year added on to the expected life span, GDP will increase by 7.7%. The variables that are negatively correlated with GDP are Unemployment, the Gini Index, natural disaster deaths and renewable energy. A 1% increase in unemployment amounts to a 2.1% decrease in GDP. A 1 unit increase in the GINI Index leads to a 2.8% decrease in Growth of GDP Per Capita. I think that the most interesting results from this study is the GINI Indexes effect on GDP Growth. From these results, my understanding is that inequality is bad for growth and a country with greater inequality will have slower GDP Growth.


Looking toward future research, one thing I want to improve to better understand my research, is to add more indpendent variables as proxies for sustainable development. The biggest difficulty with completeing this is the lack of data there is available for enviormental and social ewuity variables, like deforestation, land use, and malnourishment. These were all variables I originally planned on incoorportating my study, however the number of observations would have been compromised. I felt that inorder to get the best understanding of how sustainable development effected economic growth; it was more important to get a larger pool of observations to study.   